### PR TITLE
fix(skyrim-platform): improve event handlers safety

### PR DESCRIPTION
**Pull request by ZikkeyLS**

### CODE REVIEW

**Comment by ZikkeyLS** [comment]
@Pospelove

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp [hunk 1]
```diff
@@ -688,20 +801,25 @@ EventResult EventHandler::ProcessEvent(
     return EventResult::kContinue;
   }
 
-  auto e = CopyEventPtr(event);
+  auto activeRefFormId = event->activeRef.get()->GetFormID();
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:804]: event->activeRef.get() may be nullptr. needs check here

Use smth like event->subject ? event->subject->GetFormID() : 0;

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp [hunk 2]
```diff
@@ -688,20 +801,25 @@ EventResult EventHandler::ProcessEvent(
...
+  auto refFormId = event->ref.get()->GetFormID();
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:805]: same

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp [hunk 3]
```diff
@@ -713,31 +831,34 @@ EventResult EventHandler::ProcessEvent(
     return EventResult::kContinue;
   }
 
-  auto e = CopyEventPtr(event);
+  auto actorFormId = event->actor.get()->GetFormID();
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:834]: same

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp [hunk 4]
```diff
@@ -750,20 +871,26 @@ EventResult EventHandler::ProcessEvent(
     return EventResult::kContinue;
   }
 
-  auto e = CopyEventPtr(event);
+  auto perkId = event->perkId;
+  auto causeFormId = event->cause.get()->GetFormID();
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:875]: same

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp [hunk 5]
```diff
@@ -876,13 +1008,14 @@ EventResult EventHandler::ProcessEvent(const RE::TESResetEvent* event,
     return EventResult::kContinue;
   }
 
-  auto e = CopyEventPtr(event);
+  auto objectId = event->object->GetFormID();
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:1011]: same

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp [hunk 6]
```diff
@@ -896,13 +1029,17 @@ EventResult EventHandler::ProcessEvent(const RE::TESSellEvent* event,
     return EventResult::kContinue;
   }
 
-  auto e = CopyEventPtr(event);
+  auto sellerId = event->seller->GetFormID();
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:1032]: same

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp [hunk 7]
```diff
@@ -1656,28 +1879,29 @@ EventResult EventHandler::ProcessEvent(
     return EventResult::kContinue;
   }
 
-  auto e = CopyEventPtr(event);
+  auto worldspaceId = event->worldspaceID;
+  auto markerType = event->mapMarkerData->type.get();
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:1883]: better to check mapMarkerData for nullptr too

**Review by Pospelove — CHANGES_REQUESTED** [review]
may be I didn't see some similar cases with possible using null pointers.

please take a look carefully.

**Review by Pospelove — CHANGES_REQUESTED** [review]
I forgot to ask you to add changelog line.

smth like "fixed events handling, should be less crashes"

**Pull request merged**